packages/docs-site/blog/2023/06/29/new-language-features.md

```diff
@@ -0,0 +1,302 @@
+
+If the symmetric version of `Equal(B, A)` is considered, however, the matcher can also produce `{x -> A, y -> B}`, which would be compatible with the mapping `{y -> B, z -> C}` produced from matching `Equal(y, z)` against `Equal(B, C)`.
+
+This bug was first realized by Nimo and written up by me in [this issue](https://github.com/penrose/penrose/issues/1126) in October 2022. The [fix](https://github.com/penrose/penrose/pull/1127) involves requiring the predicate matcher to always attempt _both_ versions of the symmetric predicate:
+
+```
+if (StyName !== SubName)
+  return fail
+
+toReturn = []
+
+mapping_raw = match_raw(StyArgs, SubArgs)
+
+if success:
+  toReturn.push(mapping_raw)
+
+// flip the arguments of `SubArgs`
+SubArgsSym = [SubArgs[1], SubArgs[0]]
+
+mapping_sym = match_raw(StyArgs, SubArgsSym)
+
+if success:
+  toReturn.push(mapping_sym)
+
+if toReturn.length === 0:
+  return fail
+
+return toReturn
+
+```
+
+## Predicate Aliasing
+
+We changed the Style language to allow the Style writer to give an alias to a matched predicate so that values and shapes can be associated to that matched predicate.
+
+### Why did we need this?
+
+As an example, say we would like to draw a line between two `Atom`s whenever a `Bond` predicate exists between them. We could write,
+
+```
+forall Atom a; Atom b
+where Bond(a, b) {
+  ???.bondLine = Line {
+    -- ...
+  }
+}
+```
+
+The question is, what does `bondLine` belong to? In other words, that would `???` be? Under this circumstance, what _can_ we fill in the blank `???`?
+
+- We can either use `a` or `b`. But that would imply `bondLine` belongs to either `a` or `b`, which does not make sense - the line of a `Bond` does not belong to either of its arguments - it is shared between `a` and `b`.
+- We can leave it blank, making it `bondLine = Line { ... }`. The issue is that we can no longer refer to `bondLine` outside of the block, and so cannot override its properties elsewhere.
+
+Naturally, `bondLine` should belong to the predicate `Bond(a, b)`. It would be nice to say something to the effect of `Bond(a, b).bondLine`.
+
+### What have we done?
+
+The idea of predicate aliasing was first envisioned in July 2021 by [Helena Yang](https://heleaf.me/)'s [PR](https://github.com/penrose/penrose/pull/623) but was never merged. In July 2022, I picked up the issue in a new [PR](https://github.com/penrose/penrose/pull/1066), made it compatible with the codebase at that time, and merged it.
+
+Predicate aliasing allows the Style author to write:
+
+```
+forall Atom a; Atom b
+where Bond(a, b) as bond {
```

Comment:
Consider using [code focus](https://vitepress.dev/guide/markdown#focus-in-code-blocks) to emphasize lines